pkg/chains/storage/grafeas/grafeas_test.go

```diff
@@ -0,0 +1,382 @@
+	testListOccurrences(ctx, t, client)
+}
+
+// test attestation storage and retrieval
+func testInterface(ctx context.Context, t *testing.T, test testConfig, backend Backend, payload []byte, signature string, opts config.StorageOpts) {
+	if err := backend.StorePayload(ctx, payload, signature, opts); (err != nil) != test.wantErr {
+		t.Fatal("Backend.StorePayload() failed. error:", err, "wantErr:", test.wantErr)
+	}
+
+	// get uri
+	var objectIdentifier string
+	switch opts.PayloadFormat {
+	case formats.PayloadTypeSimpleSigning:
+		objectIdentifier = backend.getOCIURI(opts)
+	case formats.PayloadTypeInTotoIte6:
+		objectIdentifier = backend.getTaskRunURI()
+	default:
+		// for other signing formats, grafeas backend will not support
+		// we set a fake identifier for testing only
+		objectIdentifier = "placeholder_uri"
+	}
+
+	// check signature
+	expect_signature := map[string][]string{objectIdentifier: []string{signature}}
+	got_signature, err := backend.RetrieveSignatures(ctx, opts)
+	if err != nil {
+		t.Fatal("Backend.RetrieveSignatures() failed. error:", err)
+	}
+
+	if !cmp.Equal(got_signature, expect_signature) && !test.wantErr {
+		t.Errorf("Wrong signature object received, got=%s", cmp.Diff(got_signature, expect_signature))
+	}
+
+	// check payload
+	expect_payload := map[string]string{objectIdentifier: string(payload)}
+	got_payload, err := backend.RetrievePayloads(ctx, opts)
+	if err != nil {
+		t.Fatal("RetrievePayloads.RetrievePayloads() failed. error:", err)
+	}
+
+	if !cmp.Equal(got_payload, expect_payload) && !test.wantErr {
+		t.Errorf("Wrong payload object received, got=%s", cmp.Diff(got_payload, expect_payload))
+	}
+}
+
+// test occurrences are generated correctly
+func testListOccurrences(ctx context.Context, t *testing.T, client pb.GrafeasV1Beta1Client) {
+	wanted := &pb.ListOccurrencesResponse{
+		Occurrences: []*pb.Occurrence{
+			// occurrence for taskrun
+			{
+				// Occurrence Name will be automatically generated by grafeas server based on resource uri.
+				// In this fake grafeas server, we mock this behaviour by just using resource URI as the auto-generated occurrence name.
+				// In the real world, the auto-generated name will be in the format of `projects/<PROJECT_NAME>/occurrences/<AUTO-GENERATED-ID>`.
+				// i.e. projects/my_project/occurrences/06d6e0d6-ee2b-4629-b44a-2188ac92eee4
+				Name:     "/apis/tekton.dev/v1beta1/namespaces/foo1/TaskRun/bar1@uid1",
+				Resource: &pb.Resource{Uri: "/apis/tekton.dev/v1beta1/namespaces/foo1/TaskRun/bar1@uid1"},
+				NoteName: "projects/test-project/notes/test-note",
+				Details: &pb.Occurrence_Attestation{
+					Attestation: &attestationpb.Details{
+						Attestation: &attestationpb.Attestation{
+							Signature: &attestationpb.Attestation_GenericSignedAttestation{
+								GenericSignedAttestation: &attestationpb.GenericSignedAttestation{
+									// ContentType: for taskrun, this will be GenericSignedAttestation_CONTENT_TYPE_UNSPECIFIED, so this will not show up
```

Comment:
You should be able to still set this - GenericSignedAttestation_CONTENT_TYPE_UNSPECIFIED just happens to map to the default value.

Reply:
thanks!